src/openrct2/windows/top_toolbar.c

```diff
@@ -365,12 +365,13 @@ static void window_top_toolbar_mouseup(rct_window *w, sint32 widgetIndex)
 		window_news_open();
 		break;
 	case WIDX_MUTE:
-		if(gGameSoundsOff)
-			audio_unpause_sounds();
-		else
-			audio_pause_sounds();
-		break;
-	}
+		gConfigSound.sound_enabled = !gConfigSound.sound_enabled;
+		gConfigSound.ride_music_enabled = !gConfigSound.ride_music_enabled;
+		if (!gConfigSound.ride_music_enabled) {
+			audio_stop_ride_music();
+		}
+		config_save_default();
+		break;	}
```

Comment:
Wait, I now realise why you did it with audio_pause_sounds(). Sorry about that, could you revert it to how you did it first?